### ISSUE TITLE
Better centering of capsule names

### ISSUE THREAD

**Issue by peyre** [raceintospace/raceintospace#361]
(This is already accomplished; I just created this Issue to document it.)

BARIS left-aligned capsule names, among other things.  In RIS I centered them better, having the game align them based on the number of characters in the name (so if someone changed the name of a capsule it wouldn't be off center).  Letters in the game are mostly fixed-width, except for the I, so this didn't work too well for Jupiter, or especially Gemini.  So I recently added code to adjust the centering depending how many times "I" appears in the name.

Here are side-by-sides of the capsule name centering in BARIS, Sourceforge RIS, and RIS 1.2.

![Capsules1 01](https://user-images.githubusercontent.com/3118014/83654712-53dbf400-a572-11ea-9b54-04da52181db8.png)
![Capsules1 1](https://user-images.githubusercontent.com/3118014/83654716-54748a80-a572-11ea-9954-cd6707ebd9c2.png)
![Capsules1 2](https://user-images.githubusercontent.com/3118014/83654719-550d2100-a572-11ea-9a51-febdbc91ca17.png)

![Complex1 01](https://user-images.githubusercontent.com/3118014/83654711-52aac700-a572-11ea-8149-af231ecbed16.png)
![Complex1 1](https://user-images.githubusercontent.com/3118014/83654713-53dbf400-a572-11ea-90d6-8270126868ca.png)
![Complex1 2](https://user-images.githubusercontent.com/3118014/83654717-54748a80-a572-11ea-9100-07c4d3f92e94.png)

**Comment by rnyoakum** [comment]
I actually wrote some code for a recent commit that helps with this, adding a new function `int TextDisplayLength(const char *str)` to draw.h. It calculates how long, in pixels, a text string would be if displayed using the `void draw_string(int x, int y, const char *s)` function. I was going to create a text label object that would automatically handle text justification, but in the meantime you might try using `TextDisplayLength()`. I used it in rdplex.cpp(), if you want an example.

**Comment by peyre** [comment]
Oh thanks, that sounds like a much more elegant solution than mine.

**Comment by peyre** [comment]
I finally got around to checking it out--that's really cool!  I hope you don't mind, I moved it over 1px to the right to (I think) slightly better center it.

    On Wednesday, June 3, 2020, 9:58:43 AM PDT, Ryan Yoakum <notifications@github.com> wrote:  
 
 


I actually wrote some code for a recent commit that helps with this, adding a new function int TextDisplayLength(const char *str) to draw.h. It calculates how long, in pixels, a text string would be if displayed using the void draw_string(int x, int y, const char *s) function. I was going to create a text label object that would automatically handle text justification, but in the meantime you might try using TextDisplayLength(). I used it in rdplex.cpp(), if you want an example.
